Add option to specify tars to check or extract

forsyth2 opened the pull request:
Specify starting tar. Resolves #164.

forsyth2 commented on zstash/extract.py:
```diff
@@ -162,14 +189,30 @@ def extract_database(
+                (tar + ".tar",),
+            )
+            matches_ = matches_ + cur.fetchall()
+    else:  # TODO: should this always be run (i.e., not if/else?)
+        # Find matching files
+        for args_file in args.files:
+            cur.execute(
+                u"select * from files where name GLOB ? or tar GLOB ?",
+                (args_file, args_file),
+            )
+            matches_ = matches_ + cur.fetchall()
```
Should we still run this section if `--tars` is being used? That is, is there a use case where a user would specify `zstash check --hpss=<HPSS> --tars=<tars> <specific files>`? It seems like if a user is requesting specific files, it wouldn't be necessary to specify which tars to look at.

tangq replied:
That's right. When specifying the tar files, we don't need the `--tars` option - It's redundant.

forsyth2 replied:
@tangq If a user specifies both `<tars>` and `<specific files>`, what should be the expected behavior?
1. Just check/extract `<tars>`
2. Just check/extract `<specific files>`
3. Print error message saying to only specify one

tangq replied:
I'd say option 3. What do you think?

forsyth2 replied:
I would agree. @golaz which is your preference?

golaz replied:
@forsyth2 and @tangq : this is an edge case and I would think that it is unlikely a user would want to specify `<tars>` and `<specific files>` simultaneously. Therefore, if it's relatively straightforward to implement, I would suggest option 3 as well to make sure there is no room for misunderstanding from an unsuspecting user.

forsyth2 reviewed (commented):
@golaz This is ready for review.

tangq reviewed (approved):
Looks good by visual check. If testing successfully and removing the redundant `--tars` option when specifying the file names to check, it can be merged.

golaz reviewed (approved):
@forsyth2 : the code looks good. I made a few minor suggestions. Let me know if you'd like me to try it out myself.

forsyth2 commented:
@golaz @tangq I created a commit addressing your comments: https://github.com/E3SM-Project/zstash/pull/170/commits/e139de7d852ca72159bce37e0b0a7f66f47de618

Let me know if that looks good. If so, I'll squash and merge this pull request.

golaz commented:
Thanks @forsyth2. The changes look good to me.
